File: AndroidViewModel/ViewModelDataBinding/app/src/main/java/com/haoyudu/viewmodeldatabinding/MainActivity.java
package com.haoyudu.viewmodeldatabinding;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;
import com.haoyudu.viewmodeldatabinding.databinding.ActivityMainBinding;


public class MainActivity extends AppCompatActivity {
    ViewModelNum viewModelNum;
//    TextView textView;
//    Button btn;
    ActivityMainBinding binding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding=DataBindingUtil.setContentView(this, R.layout.activity_main);
        viewModelNum=ViewModelProviders.of(this).get(ViewModelNum.class);
        binding.setData(viewModelNum);
        binding.setLifecycleOwner(this);
    }
}

File: AndroidDatabase/RoomDatabaseSimpleOperate/RoomCRUD/app/src/main/java/com/haoyudu/roomcrud/MainActivity.kt
package com.haoyudu.roomcrud

import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.LinearLayout
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.recyclerview.widget.LinearLayoutManager
import com.haoyudu.roomcrud.adapter.StudentAdapter
import com.haoyudu.roomcrud.dao.StudentDao
import com.haoyudu.roomcrud.entity.Student
import kotlinx.android.synthetic.main.activity_main.*

class
MainActivity : AppCompatActivity() {
    @RequiresApi(Build.VERSION_CODES.P)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}